packages/backend/src/common/types/providers/IEmailProvider.ts

```diff
@@ -0,0 +1,8 @@
+import type { SendSmtpEmail, CreateSmtpEmail } from 'sib-api-v3-typescript';
+import type { IncomingMessage } from 'http';
+
+export interface IEmailProvider {
```

Comment:
Please use for interface custom classes. It needed to easily change the provider. Return just Promise<boolean> - either sent successfully or not. As arguments it should require only email of a user and actual message